src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -51,6 +52,32 @@ Logger frontend_logger;
 std::set<ProcessInfo> processData;
 std::mutex aggregateWeightMutex;
 std::mutex triangleTreeMutex;
+// Thread function
+void listen_to_kafka_topic(KafkaConnector &kstream,Partitioner &graphPartitioner,vector<DataPublisher*> &workerClients)
+{
+    while (true) {
+        cppkafka::Message msg = kstream.consumer.poll();
+        if (!msg || msg.get_error()) {
+            continue;
+        }
+        string data(msg.get_payload());
+        if (data == "-1") {  // Marks the end of stream
+            frontend_logger.log("Received the end of stream", "info");
+            break;
+        }
+        auto edgeJson = json::parse(data);
+        auto sourceJson = edgeJson["source"];
+        auto destinationJson = edgeJson["destination"];
+        std::string sId = std::string(sourceJson["id"]);
+        std::string dId = std::string(destinationJson["id"]);
+        partitionedEdge partitionedEdge = graphPartitioner.addEdge({sId, dId});
+        sourceJson["pid"] = partitionedEdge[0].second;
+        destinationJson["pid"] = partitionedEdge[1].second;
+        workerClients.at((int) partitionedEdge[0].second)->publish(sourceJson.dump());
+        workerClients.at((int) partitionedEdge[1].second)->publish(destinationJson.dump());
+    }
+    graphPartitioner.printStats();
```

Comment:
What kind of stats do we print here? If we do this for every data item we receive, that will be unnecessary overhead.

Reply:
The following stats will print after termination of kafka stream but not every time.

![Screenshot from 2022-10-23 16-37-52](https://user-images.githubusercontent.com/62076661/204126099-7eb75868-e17f-4863-a996-35bbc2cdf533.png)